New popcorn

Major0911 opened the pull request:
Two new types of popcorn have been added - salted and with caramel.

## About The Pull Request

Two new types of popcorn have been added, which are created by adding salt or caramel to the ready-made, simple popcorn. Also added empty versions of popcorn packs as garbage after them.

## Why It's Good For The Game

Old popcorn has already bored many people, both in appearance and because it is, literally, tasteless popcorn. Who would eat that? Now the station staff will have access to the most popular types of this snack - salty and sweet. Just like regular popcorn, new types are eaten for a long time, from which the original idea remains valid

## Changelog
:cl:
add: new popcorn flavors and two kinds of empty boxes after them
/:cl:

Major0911 committed:
New popcorn

Two new types of popcorn have been added - salted and with caramel.

etherware-novice commented:
woo good job man

Mothblocks commented on code/game/objects/items/food/misc.dm:
```diff
@@ -939,3 +939,29 @@
 	foodtypes = GRAIN | VEGETABLES
 	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/food/caramel_popcorn
```
These should probably be subtypes of popcorn so we do not need to copy and paste some of these properties (like bite-consumption).

Major0911 replied:
Hm... Perhaps it could have been done this way. Although, since this was my first published work, I tried not to make serious mistakes, and this could lead to them

Mothblocks replied:
You resolved this but did not do it

Mothblocks commented on code/game/objects/items/food/misc.dm:
```diff
@@ -939,3 +939,29 @@
+/obj/item/food/caramel_popcorn
+	name = "caramel popcorn"
+	icon_state = "сaramel_popcorn"
+	desc = "caramel-covered popcorn. Sweet!"
```
```suggestion
	desc = "Caramel-covered popcorn. Sweet!"
```

Mothblocks commented on code/game/objects/items/trash.dm:
```diff
@@ -153,3 +153,13 @@
+
+/obj/item/trash/caramel_popcorn
+	name = "empty caramel popcorn"
+	desc = "now it's not a sweet snack, but just a sticky bag..."
```
```suggestion
	desc = "Now it's not a sweet snack, but just a sticky bag..."
```

Mothblocks commented on code/modules/food_and_drinks/recipes/tablecraft/recipes_misc.dm:
```diff
@@ -324,3 +324,21 @@
+	name ="Caramel popcorn"
+	reqs = list(
+		/obj/item/food/popcorn = 1,
+		/datum/reagent/consumable/caramel = 3
```
```suggestion
		/datum/reagent/consumable/caramel = 3,
```

Helps keep blames looking nice if more ingredients are added.

Mothblocks commented on code/modules/food_and_drinks/recipes/tablecraft/recipes_misc.dm:
```diff
@@ -324,3 +324,21 @@
 	subcategory = CAT_MISCFOOD
+
+/datum/crafting_recipe/food/caramel_popcorn
+	name ="Caramel popcorn"
```
```suggestion
	name = "Caramel popcorn"
```

Mothblocks commented on code/modules/food_and_drinks/recipes/tablecraft/recipes_misc.dm:
```diff
@@ -324,3 +324,21 @@
+	subcategory = CAT_MISCFOOD
+
+/datum/crafting_recipe/food/salty_popcorn
+	name ="Salty popcorn"
```
```suggestion
	name = "Salty popcorn"
```

Mothblocks commented on code/modules/food_and_drinks/recipes/tablecraft/recipes_misc.dm:
```diff
@@ -324,3 +324,21 @@
+	name ="Salty popcorn"
+	reqs = list(
+		/obj/item/food/popcorn = 1,
+		/datum/reagent/consumable/salt = 3
```
```suggestion
		/datum/reagent/consumable/salt = 3,
```

Mothblocks reviewed (changes requested):
It's curious to me that janitor.dmi size went *down*? Icon diff bot doesn't seem to suggest anything is amiss, so probably not worth worrying about.

optimumtact commented:
This is funny because it implies everyone has just been eating plain popcorn like pyschos for the last 11 years

Major0911 committed:
Update code/game/objects/items/food/misc.dm

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

Major0911 committed:
Update code/game/objects/items/trash.dm

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

Major0911 committed:
Update code/modules/food_and_drinks/recipes/tablecraft/recipes_misc.dm

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

Major0911 committed:
Update code/modules/food_and_drinks/recipes/tablecraft/recipes_misc.dm

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

Major0911 committed:
Update code/modules/food_and_drinks/recipes/tablecraft/recipes_misc.dm

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

Major0911 committed:
Update code/modules/food_and_drinks/recipes/tablecraft/recipes_misc.dm

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

ShizCalev commented on code/game/objects/items/food/misc.dm:
```diff
@@ -939,3 +939,29 @@
+	tastes = list("caramel" = 2, "popcorn" = 1)
+	foodtypes = JUNKFOOD | SUGAR
+	food_flags = FOOD_FINGER_FOOD
+	w_class = WEIGHT_CLASS_SMALL
```
Make these subtypes of /popcorn. You'll get rid of 4 vars each and fix the fact you're not using the right eat verbs.

Mothblocks replied:
oh yeah i asked them to do this earlier and forgot i asked

Major0911 committed:
New popcorn

The structure of the new files has been slightly changed in favor of convenience.

ShizCalev commented on code/game/objects/items/food/misc.dm:
```diff
@@ -939,3 +956,7 @@
 	foodtypes = GRAIN | VEGETABLES
 	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/food/caramel_popcorn
```
```suggestion
```

ShizCalev committed:
Update code/game/objects/items/food/misc.dm

ShizCalev commented on code/game/objects/items/trash.dm:
```diff
@@ -153,3 +153,13 @@
 	desc = "It's been Donk-decimated."
 	icon_state = "ready_donk"
+
+/obj/item/trash/caramel_popcorn
```
subtypes please.

Major0911 committed:
New popcorn

Minor code fixes

Major0911 committed:
Merge branch 'popcorn-item' of https://github.com/Major0911/Sprites-and-other into popcorn-item
